hashintel/hash
# Check console messages in Playwright

## 🌟 What is the purpose of this PR?

This PR follows #1344 and configures Playwright to check unexpected console messages. This saves us from accidentally merging PRs that clutter the console.



## ⚠️ Known issues



## 🚫 Blocked by




- [x] https://github.com/hashintel/hash/pull/1344

## 🐾 Next steps

- Re-enable more tests
- Remove these unexpected messages:

  ```
  Array [
    "[error] Failed to load resource: the server responded with a status of 401 (Unauthorized)",
    "[error] Failed to load resource: the server responded with a status of 401 (Unauthorized)",
  ]
  ```

  (to be addressed after backend refactoring)

## 🔍 What does this change?




- Unexpected console messages (logs, warnings or errors) fail tests. Idea source: https://github.com/microsoft/playwright/discussions/11690#discussioncomment-2060397. The tests still run till the end and throw after all other assertions.

- Test developers need to import from `./shared/runtime` instead of `@playwright/test` for the new check to work.

- [Autocomplete attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) are added to the temp login and signup forms. This avoids:

  ```
     "[verbose] [DOM] Input elements should have autocomplete attributes (suggested: \"username\"): (More info: https://goo.gl/9p2vKq) %o",
  ```
  (the error only shows locally during `yarn dev`)

### 📜 Does this require a change to the docs?



- No. New ESLint rule makes sure that the correct import is used in tests.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200211978612931/1202027529382116/f) _(internal)_
- [Slack thread](https://hashintel.slack.com/archives/C022217GAHF/p1648219678107789) _(internal)_

## 🛡 What tests cover this?



- CI

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Add `console.log` somewhere
1.  Run the app
1.  Run `yarn test:playwright`
1.  Observe test failure

## 📹 Demo

<img width="998" alt="Screenshot 2022-11-04 at 18 30 52" src="https://user-images.githubusercontent.com/608862/200049513-77308eb4-e62d-4504-a2dd-9896439e7917.png">
<img width="1012" alt="Screenshot 2022-11-04 at 18 30 41" src="https://user-images.githubusercontent.com/608862/200049521-5f2126d4-548e-4fdb-ab7b-f7e3dfd6d421.png">
